### DISCUSSION TITLE
Http Client Cookies

### DISCUSSION THREAD

**Discussion by gam04** [laravel/framework#43578] [Q&A]
Hi There 😄 .

I'm using the HTTP Client Facade. I'm traying to use cookies, it is possible to set multiples cookies using `withCookies` method? :  

```php
  $getResponse = Http::withHeaders($this->headers)->get($url);
  $cookies = $getResponse->cookies(); // Instance of Guzzle/CookieJar
  
  // $cookies->toArray() returns an array of arrays, multiples cookies
   $postResponse = Http::withHeaders($this->headers)
                ->withCookies($cookies->toArray(), $domain)
                ->post($url, [
                    'FORM_planes' => 'FORM_planes',
                    'FORM_planes:BTN_planPublico1' => '',
                    'javax.faces.ViewState' => $viewState
    ]);
  
```

By this way i get `Array to string conversion Exception` . It seems the method expect just one cookie.

Thanks in advance.

Regards.

**Comment by gam04** [comment]
Sorry, I jus saw the issue, pls ignore this discussion, thanks. 😄

**Reply by AbdelkaderGailany** [reply]
What is the solution for this problem

**Reply by henzeb** [reply]
The cookiejar gives a very different result on toArray than the 'withCookies' method accepts. Be sure the withCookies receives an array of the format [name=>value] (same with CookieJar::fromArray, which is what 'withCookies' is accepting.

**Reply by bilogic** [reply]
Is there a built in way to convert the cookies from the 1st response and pass it to the 2nd request?

**Reply by Kenny1291** [reply]
> Is there a built in way to convert the cookies from the 1st response and pass it to the 2nd request?

~You can set up a shared cookie jar for each request: [Guzzle docs](https://docs.guzzlephp.org/en/stable/quickstart.html#cookies)~

~$client = Http::buildClient(['cookies' => true]);~

```php 
$client = Http::buildClient();
```
Then use that client for every request:
```php
Http::setClient($client)->post() ...
```
EDIT: As @ZacharyDuBois pointed out `['cookies' => true]` is irrelevant.

**Reply by ZacharyDuBois** [reply]
@Kenny1291 `Http::buildClient()` doesn't accept any arguments. It produces a fully new instance of a Guzzle client with the Laravel middleware. Another side effect of this pattern, it won't track with Laravel Telescope (appears the events aren't firing). One of the APIs I interact with for work requires the client to maintain a session cookie for each parallel request (each session can only have one concurrent request). I would like the nice insights from Sentry / Telescope for these requests but it seems there is not a decent way to get this working with the Laravel HTTP Client.

Anyone else have any ideas? Something like https://github.com/laravel/ideas/issues/2479 would be awesome. Would allow people to make a wrapper around the Laravel HTTP client while maintaining cookies.
